src/utils/index.js

```diff
@@ -279,99 +307,111 @@ export const isItemAFarmProduct = item =>
       item.type === itemType.CRAFTED_ITEM
   )
 
-/**
- * @param {farmhand.cropTimetable} cropTimetable
- * @returns {Array.<enums.cropLifeStage>}
- */
-export const getLifeStageRange = memoize(cropTimetable =>
+export const getLifeStageRange = memoize((
+  /** @type {farmhand.cropTimetable} */ cropTimetable
+) =>
   [SEED, GROWING].reduce(
+    /**
+     * @param {farmhand.cropLifeStage[]} acc
+     */
     (acc, stage) => acc.concat(Array(cropTimetable[stage]).fill(stage)),
     []
   )
 )
 
 /**
  * @param {farmhand.crop} crop
- * @returns {enums.cropLifeStage}
+ * @returns {farmhand.cropLifeStage}
  */
-export const getCropLifeStage = ({ itemId, daysWatered }) =>
-  getLifeStageRange(itemsMap[itemId].cropTimetable)[Math.floor(daysWatered)] ||
-  GROWN
+export const getCropLifeStage = crop => {
+  const { itemId, daysWatered } = crop
+  const { cropTimetable } = itemsMap[itemId]
+
+  if (!cropTimetable) {
+    throw new Error(`${itemId} has no cropTimetable`)
+  }
+
+  return getLifeStageRange(cropTimetable)[Math.floor(daysWatered)] || GROWN
+}
 
 /**
- * @param {farmhand.plotContent} plotContent
+ * @param {farmhand.plotContent | farmhand.shoveledPlot | null} plotContents
  * @param {number} x
  * @param {number} y
  * @returns {?string}
  */
-export const getPlotImage = (plotContent, x, y) => {
-  if (plotContent) {
-    if (getPlotContentType(plotContent) === itemType.CROP) {
+export const getPlotImage = (plotContents, x, y) => {
```

Comment:
This function needed the most refactoring, but it should be functionally the same as before.